alacritty/src/renderer/text/builtin_font.rs

```diff
@@ -500,106 +506,50 @@ fn box_drawing(character: char, metrics: &Metrics, offset: &Delta<i8>) -> Raster
-    // Use one eight of the cell width, since this is used as a step size for block elements.
-    let stroke_size = cmp::max((width as f32 / 8.).round() as usize, 1) as f32;
+    let extra_thickness = calculate_stroke_size(width) as i32 - 1;
 
     let mut canvas = Canvas::new(width, height);
 
-    let y_center = (height - 1) as f32 / 2.;
-    // Start with offset `1` and draw until the intersection of the f(x) = x + 1 and
-    // g(x) = H - x + 1 lines. The intersection happens when f(x) = g(x), which is at
-    // x = H/2 (`y_center`).
-    let from_y = 1;
-    let x_end = y_center.floor();
-    let y_end = (height - from_y - 1) as f32;
-
-    // Pick the start point outside of the canvas to even-out the start.
-    let from_x = 0.;
-    let to_x = x_end;
-    canvas.draw_line_grid(from_x, from_y as f32, to_x, y_center.floor());
-    canvas.draw_line_grid(from_x, y_end, to_x, y_center.ceil());
-
-    // For regular arrows we handle thickness by drawing 2 angle arrows and then just filling
-    // the contents between them.
-    if (character == '\u{e0b1}' || character == '\u{e0b3}') && stroke_size > 1. {
-        // The default line is of stroke size 1, so the 0.5 is computed by subtracting 1 from
-        // stroke_size and then adding 0.5 to to put the target in the center of the cell.
-        let to_x = x_end - stroke_size;
-        canvas.draw_line_grid(from_x, from_y as f32 + stroke_size, to_x, y_center.floor());
-        canvas.draw_line_grid(from_x, y_end - stroke_size, to_x, y_center.ceil());
-    }
+    let slope = 1;
+    let f_y0 = 1;
+    let g_y0 = height as i32 - f_y0 - 1;
 
-    let buffer = canvas.buffer_mut();
-    if character == '\u{e0b0}' || character == '\u{e0b2}' {
-        for row in from_y..height - from_y {
-            let row_offset = row * width;
-            for index in 1..width {
-                let index = row_offset + index;
-                if buffer[index - 1]._r > buffer[index]._r && buffer[index]._r == 0 {
-                    break;
-                }
-
-                buffer[index - 1] = COLOR_FILL;
-            }
-        }
-    } else if stroke_size > 1. {
-        // Find the bottom/top most points of extra line we draw, so we can properly set the
-        // `start`.
-
-        let mut y1 = 0;
-        for row in (0..height / 2).rev() {
-            if buffer[row * width]._r != 0 {
-                y1 = row;
-                break;
-            }
-        }
-        let mut y2 = height / 2;
-        for row in height / 2..height {
-            if buffer[row * width]._r != 0 {
-                y2 = row;
+    // Start with offset `1` and draw until the intersection of the f(x) = x + 1 and
+    // g(x) = H - x - 1 lines. The intersection happens when f(x) = g(x), which is at
+    // x = H/2 - 1.
+    let intersection = (height as i32 + 1) / 2 - 1;
+
+    let f_x = LineEquation::new(slope, f_y0, 0, intersection);
+    let g_x = LineEquation::new(-slope, g_y0, 0, intersection);
```

Comment:
Honestly thinking about it `f_x` and `g_x` are pretty terrible names. Would be much easier to understand all of this by just naming them after what they actually are. Something like `top_line` and `bottom_line` maybe.

The whole `_x` thing is just confusing because it's basically referring to a variable that isn't actually passed to the line function (we just call `.next()`). It makes sense mathematically but the concept just doesn't translate.